src/MoBi.Core/Serialization/ORM/Mappers/ProjectMetaDataToProjectMapper.cs

```diff
@@ -91,9 +92,22 @@ private void addNonProjectBuildingBlocks(List<IBuildingBlock> moduleBuildingBloc
          var simulationSettingsBlocks = moduleBuildingBlocks.OfType<SimulationSettings>().ToList();
          simulationSettingsBlocks.Each(x => moduleBuildingBlocks.Remove(x));
 
-         addModuleBuildingBlocks(moduleBuildingBlocks);
+
+         var uniqueBuildingBlockTypes = hasUniqueBuildingBlocks(moduleBuildingBlocks);
+         if(uniqueBuildingBlockTypes)
+            addModuleBuildingBlocks(moduleBuildingBlocks);
+         else
+            addBuildingBlocksToModule(moduleBuildingBlocks);
+         
          addSimulationSettingsBuildingBlocks(simulationSettingsBlocks);
       }
+      private void addBuildingBlocksToModule(List<IBuildingBlock> moduleBuildingBlocks)
+      {
+         if (!moduleBuildingBlocks.Any())
```

Comment:
Is that needed? I think the `Each` implies that if there are none, then nothing will happen